lib/rubocop/cop/lint/inline_access_modifier.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # Access modifiers should be inlined in method definitions for clarity.
+      #
+      # @example
+      #
+      #   # bad
+      #
+      #   class Foo
+      #
+      #     private
+      #
+      #     def baz; end
+      #     def foo; end
+      #
+      #   end
+      #
+      # @example
+      #
+      #   # good
+      #
+      #   class Foo
+      #
+      #     private def baz; end
+      #     private def foo; end
+      #
+      #   end
+      class InlineAccessModifier < Cop
+        MSG = '`%s` should be inlined in method definitions for clarity.'.freeze
+
+        def on_send(node)
+          return unless node.access_modifier?
+
+          return if access_modifier_is_inlined?(node)
+
+          add_offense(node)
+        end
+
+        private def access_modifier_is_inlined?(node)
```

Comment:
Currently, inlined access qualifiers are not used in production code of RuboCop.

```console
% git grep 'private def ' lib
lib/rubocop/ast/node/mixin/method_dispatch_node.rb:      #   private def foo; end
lib/rubocop/cop/layout/def_end_alignment.rb:      #   private def foo
lib/rubocop/cop/layout/def_end_alignment.rb:      #   private def foo
lib/rubocop/cop/layout/def_end_alignment.rb:      #   private def foo
lib/rubocop/cop/layout/def_end_alignment.rb:      #   private def foo
lib/rubocop/cop/layout/else_alignment.rb:            parent.loc.selector # For example "private def ..."
```

By making it consistent with existing usage, it may be able to reduce surprises.

Reply:
I have the cop disabled by default—I'd be happy to enable it and fix any offenses in the RuboCop codebase, but I didn't want to do that unless one of the maintainers is 👍 on the idea.

Reply:
Do we want to make this enable by default? I am not that sure... shouldn't this be discuss in an issue? I think there is no issue for this... 🤔

Reply:
Yeah, I didn't open an issue for this. I figured the discussion around whether or not it should be enabled by default would be simple enough that we could just discuss it in the PR.

Reply:
> I figured the discussion around whether or not it should be enabled by default would be simple enough that we could just discuss it in the PR.

Since this is by far the least favoured style (because of historical reasons), there is a huge potential backlash. Probably want to take it through the Ruby Style Guide repo first. In the meanwhile, would be nice to just use the existing convention internally. 🙂

Reply:
(Also we don't have proper handling for this kind of modifier for a lot of alignment cops, which would lead to a pretty bad cascade of issues.)